src/main/java/com/iexec/core/task/TaskController.java

```diff
@@ -110,9 +155,30 @@ public ResponseEntity<TaskStdout> getTaskStdout(@PathVariable("chainTaskId") Str
     @GetMapping("/tasks/{chainTaskId}/replicates/{walletAddress}/stdout")
     public ResponseEntity<ReplicateStdout> getReplicateStdout(
             @PathVariable("chainTaskId") String chainTaskId,
-            @PathVariable("walletAddress") String walletAddress) {
+            @PathVariable("walletAddress") String walletAddress,
+            @RequestHeader("Authorization") String bearerToken) {
+        String requesterAddress = jwtTokenProvider.getWalletAddress(bearerToken);
+        if (requesterAddress.isEmpty()) {
+            return ResponseEntity.status(HttpStatus.UNAUTHORIZED).build();
+        }
+        if (!isStdoutRequesterSameAsTaskRequester(requesterAddress, chainTaskId)) {
```

Comment:
This could work for task but would be problematic for replicate where `walletAddress` is already defined for the worker address.
I would prefer to keep a method name which is explicit on what is checked.
What do you think of `isTaskRequester(stdoutRequester, chainTaskId)` ?

Reply:
I see, your suggestion is good yes :+1:

Reply:
see 5c51c87